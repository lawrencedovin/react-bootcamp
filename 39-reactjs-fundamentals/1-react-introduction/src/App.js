import logo from './logo.svg';
import MovieList from './components/MovieList/MovieList';
import Dog from './components/Dog/Dog';
import './App.css';

function App() {
  return (
    <div className="App">
      <MovieList />
      <Dog src="https://images.unsplash.com/photo-1550035605-144be67b33ea?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80" />
      <Dog src="https://images.unsplash.com/photo-1619979842913-08603df442de?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80" />
    </div>
  );
}

export default App;
