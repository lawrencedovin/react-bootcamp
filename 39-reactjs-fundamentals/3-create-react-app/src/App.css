.shopping__logo {
    height: 50px;
}