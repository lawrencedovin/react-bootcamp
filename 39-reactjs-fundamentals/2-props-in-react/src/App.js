import logo from './logo.svg';
import './App.css';
import Animal from './components/Animal/Animal';
import RandomChoice from './components/RandomChoice/RandomChoice';
import CoinFlip from './components/CoinFlip/CoinFlip';
import TodosList from './components/TodosList/TodosList';

const todos = [
  {"id": 1, "name": "buy burger"},
  {"id": 2, "name": "eat burger"},
  {"id": 3, "name": "celebrate"},
  {"id": 4, "name": "praise the LORD!"},
]

const App = () => (
    <div className="App">
      <Animal emoji="🦁" name="Lion" sound="roar" />
      <Animal emoji="🐶" name="Dog" sound="bark" />
      <Animal emoji="😺" name="Cat" sound="meow" />
      <RandomChoice array={['red', 'green', 'yellow']} />
      <CoinFlip />
      <TodosList array={todos} />
    </div>
)

export default App;
