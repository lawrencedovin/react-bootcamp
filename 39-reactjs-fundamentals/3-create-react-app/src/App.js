import './App.css';
import ShoppingList from './ShoppingList/ShoppingList';
import items from './items';

function App() {
  return (
    <>
      <ShoppingList items={items} />
    </>
  );
}

export default App;
