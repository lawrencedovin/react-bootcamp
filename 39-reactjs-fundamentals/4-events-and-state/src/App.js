import logo from './logo.svg';
import './App.css';
import Clicker from './components/Clicker';
import ButtonGroup from './components/ButtonGroup';
import Counter from './components/Counter';

function App() {
  return (
    <div className="App">
      <Clicker message="Hello world!" btnText="Click Hello" />
      <Clicker message=":)" btnText="Click Smiley" />
      <ButtonGroup />
      <Counter />
    </div>
  );
}

export default App;
