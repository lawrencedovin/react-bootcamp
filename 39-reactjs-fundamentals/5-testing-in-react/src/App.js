import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      Hello, I'm an App!
    </div>
  );
}

export default App;
