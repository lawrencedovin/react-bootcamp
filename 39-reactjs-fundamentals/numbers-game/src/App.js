import logo from './logo.svg';
import './App.css';
import ClickRandomNumber from './components/ClickRandomButton';

function App() {
  return (
    <div className="App">
      <ClickRandomNumber maxNumber={4} />
    </div>
  );
}

export default App;
