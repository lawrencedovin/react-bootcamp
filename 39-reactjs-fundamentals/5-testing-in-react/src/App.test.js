import { render, screen } from '@testing-library/react';
import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });


// Smoke Tests
it("renders without crashing", () => {
  render(<App />);
});

// Snapshot Tests
it("matches snapshop", () => {
  const {asFragment} = render(<App />);
  expect (asFragment()).toMatchSnapshot();
});