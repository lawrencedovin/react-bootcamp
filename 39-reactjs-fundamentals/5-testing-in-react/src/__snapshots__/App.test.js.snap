// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshop 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    Hello, I'm an App!
  </div>
</DocumentFragment>
`;
